homeassistant/components/palazzetti/climate.py

```diff
@@ -0,0 +1,164 @@
+"""Support for Palazzetti climates."""
+
+from typing import Any, cast
+
+from palazzetti_sdk_local_api.exceptions import InvalidStateTransitionError
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    ACTION_NOT_UNAVAILABLE,
+    API_HW_VERSION,
+    API_NAME,
+    API_SW_VERSION,
+    AVAILABLE,
+    DOMAIN,
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_MODE,
+    FAN_MODES,
+    FAN_SILENT,
+    HEATING_STATUSES,
+    MAC,
+    MODE,
+    PALAZZETTI,
+    ROOM_TEMPERATURE,
+    TARGET_TEMPERATURE,
+)
+from .coordinator import PalazzettiDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Palazzetti climates based on a config entry."""
+    coordinator: PalazzettiDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    entities: list[PalazzettiClimateEntity] = []
+    await coordinator.async_config_entry_first_refresh()
+    if coordinator.data[AVAILABLE]:
+        entities.append(PalazzettiClimateEntity(coordinator=coordinator))
+        async_add_entities(entities)
+
+
+class PalazzettiClimateEntity(
+    CoordinatorEntity[PalazzettiDataUpdateCoordinator], ClimateEntity
+):
+    """Defines a Palazzetti climate."""
+
+    _attr_has_entity_name = True
+    _attr_hvac_modes = []  # The available modes will be set when we know the current state
+    _attr_fan_modes = FAN_MODES
+    _attr_target_temperature_step = 1.0
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_translation_key = DOMAIN
+
+    def __init__(
+        self,
+        *,
+        coordinator: PalazzettiDataUpdateCoordinator,
+    ) -> None:
+        """Initialize Palazzetti climate."""
+        super().__init__(coordinator=coordinator)
+        self.hub = coordinator.hub
+
+        # TURN_OFF and TURN_ON are not always available. An update coordinator listener will add them when possible.
+        self._attr_supported_features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.FAN_MODE
+        )
+        coordinator.async_add_listener(self._coordinator_update)
```

Comment:
```suggestion
        config_entry.async_on_unload(coordinator.async_add_listener(self._coordinator_update))
```
make sure it's removed

Reply:
I removed the listener in the new revision. Using properties directly instead.